src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -35,7 +35,7 @@ internal sealed class SqlTriggerListener<T> : IListener
         private readonly ITriggeredFunctionExecutor _executor;
         private readonly ILogger _logger;
 
-        private readonly IDictionary<TelemetryPropertyName, string> _telemetryProps;
+        private readonly IDictionary<TelemetryPropertyName, string> _telemetryProps = new Dictionary<TelemetryPropertyName, string>();
```

Comment:
Would suggest instantiating a new dictionary every time inside `StartAsync` method. It is possible that the WebJobs SDK calls `StartAsync` while the previously called `StartAsync` method is still running. We have checks to ensure proper functionality for such cases. See line 82 in this revision. With current code, we can have a situation where the second method instance clears the dictionary while the first instance (including class `SqlTableChangeMonitor`) is still logging messages.

Reply:
I'm kind of surprised that's expected behavior - what scenario does that happen in?

And instead I'm just going to move the telemetry calls below so they're only called when it's actually successfully starting. We don't want to be tracking listener starts when it doesn't actually start.

Reply:
(I did this in my open PR for additional debug logging - leave a comment there if you see any other issues)

Reply:
> I'm kind of surprised that's expected behavior - what scenario does that happen in?

I am making a guess by looking at the WebJobs SDK source code: If there are multiple listeners registered in the function app, the [`CompositeListener` will call `StartAsync` on all listeners](https://github.com/Azure/azure-webjobs-sdk/blob/09225716dce117fa2b8b14dfb0b82d2ac8e77e77/src/Microsoft.Azure.WebJobs.Host/Listeners/CompositeListener.cs#L34-L39)). If either one of the listeners throw exception, it seems [all the listeners will be called again](https://github.com/Azure/azure-webjobs-sdk/blob/09225716dce117fa2b8b14dfb0b82d2ac8e77e77/src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListener.cs#L81-L89). The listeners which have already started are supposed to be skipping performing the startup operation in this case.

Reply:
Gotcha, thanks for digging into that!